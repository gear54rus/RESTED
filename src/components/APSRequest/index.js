import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field, Fields, formPropTypes } from 'redux-form';
import { Panel, Form, Col, FormGroup, Checkbox, Alert } from 'react-bootstrap';
import flow from 'lodash.flow';

import BasicAuthSection from 'components/Request/BasicAuthSection';
import * as apsActions from 'store/aps/actions';
import { getError } from 'store/aps/selectors';
import { DEFAULT_APS_REQUEST } from 'constants/constants';

import Titlebar from './Titlebar';
import URLField from './URLField';
import TokenParamsField from './TokenParamsField';
import SubmitButton from './SubmitButton';

export const apsRequestForm = 'apsRequest';

function APSRequest({ tokenFetchError, handleSubmit, sendRequest }) {
  return (
    <Panel header={<Titlebar />}>
      <Form
        horizontal
        onSubmit={handleSubmit(sendRequest)}
      >
        <FormGroup>
          <Col
            xs={5}
          >
            <Field
              name="url"
              component={URLField}
            />
          </Col>

          <Col
            xs={7}
          >
            <Fields
              names={['username', 'password']}
              component={BasicAuthSection}
              small
            />
          </Col>
        </FormGroup>
        <FormGroup>
          <Col
            xs={7}
          >
            <Field
              name="type"
              component={TokenParamsField}
            />
          </Col>

          <Col
            xs={2}
          >
            <Checkbox
              disabled // to be implemented
            >
              Refresh automatically?
            </Checkbox>
          </Col>

          <Col
            xs={3}
          >
            <SubmitButton />
          </Col>
        </FormGroup>
        {tokenFetchError && (
          <Alert bsStyle="danger">
            <strong>Failed to fetch token!</strong> {tokenFetchError.message}
          </Alert>
        )}
      </Form>
    </Panel>
  );
}

APSRequest.propTypes = {
  ...formPropTypes,
  tokenFetchError: PropTypes.instanceOf(Error),
};

const formOptions = {
  form: apsRequestForm,
};

const mapStateToProps = state => ({
  initialValues: DEFAULT_APS_REQUEST,
  tokenFetchError: getError(state),
});

export default flow(
  reduxForm(formOptions),
  connect(mapStateToProps, {
    ...apsActions,
  }),
)(APSRequest);
