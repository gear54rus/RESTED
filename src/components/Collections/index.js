import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Panel } from 'react-bootstrap';

import IconButton from 'components/IconButton';
import * as Actions from 'store/collections/actions';

import CollectionList from './CollectionList';

function Titlebar({ addCollection }) {
  return (
    <span className="clearfix">
      <IconButton
        onClick={() => addCollection()}
        tooltip="Add new collection"
        icon="plus"
        className="pull-right"
      />
    </span>
  );
}

Titlebar.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

function Collections({ addCollection }) {
  return (
    <Panel>
      <Panel.Heading>
        <Panel.Title
          componentClass={Titlebar}
          addCollection={addCollection}
        />
      </Panel.Heading>
      <Panel.Body>
        <CollectionList />
      </Panel.Body>
    </Panel>
  );
}

Collections.propTypes = {
  addCollection: PropTypes.func.isRequired,
};

export default connect(null, Actions)(Collections);

