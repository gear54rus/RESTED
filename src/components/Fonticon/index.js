import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Fonticon({ icon, className }) {
  return (
    <i
      className={classNames('fas', `fa-${icon}`, className)}
      role="presentation"
    />
  );
}

Fonticon.propTypes = {
  icon: PropTypes.string.isRequired,
  className: PropTypes.string,
};

export default Fonticon;
