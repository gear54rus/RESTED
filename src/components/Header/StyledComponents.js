import styled, { css } from 'styled-components';

/* eslint-disable import/prefer-default-export */
export const StyledHeader = styled.header`
  margin-top: 20px;
  margin-bottom: 30px;

  img {
    margin-top: -7px;
    margin-right: 10px;
  }

  h1, a {
    color: black;
  }

  ${props => props.darkMode && css`
    h1, a {
      color: white;
    }
    
    img {
      filter: invert(100%);
    }
  `}
`;
