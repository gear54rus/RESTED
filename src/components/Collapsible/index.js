import React, { PropTypes } from 'react';
import classNames from 'classnames';
import { connect } from 'react-redux';
import { Collapse, Row, Col } from 'react-bootstrap';

import * as Actions from 'store/config/actions';
import { isOpen } from 'store/config/selectors';

import { CollapsibleButton, CollapsibleHeader, AnimatedIcon } from './StyledComponents';

export function Collapsible(props) {
  const {
    id,
    title,
    open,
    children,
    toggleCollapse,
    mountOnEnter,
    unmountOnExit,
  } = props;
  return (
    <div>
      <CollapsibleButton
        bsStyle="link"
        onClick={e => {
          e.preventDefault();
          toggleCollapse(id, open);
        }}
      >
        <CollapsibleHeader>
          {title}
          <AnimatedIcon
            icon="angle-right"
            className={classNames({
              'fa-rotate-90': open,
            })}
          />
        </CollapsibleHeader>
      </CollapsibleButton>
      <Collapse
        in={open}
        mountOnEnter={mountOnEnter}
        unmountOnExit={unmountOnExit}
      >
        <Row>
          <Col xs={12}>
            {children}
          </Col>
        </Row>
      </Collapse>
    </div>
  );
}

Collapsible.defaultProps = {
  mountOnEnter: false,
  unmountOnExit: false,
  open: false,
};

Collapsible.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
  toggleCollapse: PropTypes.func.isRequired,
  mountOnEnter: PropTypes.bool,
  unmountOnExit: PropTypes.bool,
  open: PropTypes.bool,
};

const mapStateToProps = (state, props) => ({
  open: isOpen(state, props),
});

export default connect(mapStateToProps, Actions)(Collapsible);

