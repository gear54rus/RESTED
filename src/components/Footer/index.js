import React from 'react';

import { StyledFooter } from './StyledComponents';

export default function Footer() {
  return (
    <StyledFooter>
      <small>
        <a
          href="https://github.com/odin-public/RESTED-APS"
          target="_blank"
          rel="noopener noreferrer"
        >
          GitHub
        </a>
      </small>
      ·
      <small>
        <a
          href="https://github.com/odin-public/RESTED-APS/issues"
          target="_blank"
          rel="noopener noreferrer"
        >
          Report a bug
        </a>
      </small>
      ·
      <small>
        <a
          href="https://gitter.im/RESTEDclient"
          target="_blank"
          rel="noopener noreferrer"
        >
          Chat (RESTED)
        </a>
      </small>
      ·
      <small>
        <a
          href="https://gist.github.com/esphen/f50cc049c8ed1256a11763826dac97db"
          target="_blank"
          rel="noopener noreferrer"
        >
          Licenses (RESTED)
        </a>
      </small>
    </StyledFooter>
  );
}
