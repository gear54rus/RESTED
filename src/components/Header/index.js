import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { isDarkTheme } from 'store/options/selectors';

import { StyledHeader, BlackLink } from './StyledComponents';

export function Header({ darkMode }) {
  return (
    <StyledHeader darkMode={darkMode}>
      <h1>
        <BlackLink
          href="https://github.com/gear54rus/RESTED-APS"
          target="_blank"
          rel="noopener noreferrer"
        >
          <img
            className="logo"
            role="presentation"
            height="40"
            src="img/rested-aps-logo.png"
          />
          <span>RESTED APS</span>
        </BlackLink>
      </h1>
      <a
        href="https://chrome.google.com/webstore/detail/rested-aps/omkndfeccmeplaimlpaefimnimmniccl"
        target="_blank"
        rel="noopener noreferrer"
      >
        Also available for Google Chrome{' '}
        <img
          src="https://upload.wikimedia.org/wikipedia/commons/a/a5/Google_Chrome_icon_%28September_2014%29.svg"
          style={{
            height: '2em',
          }}
          alt="Chrome icon"
        />
      </a>
    </StyledHeader>
  );
}

Header.propTypes = {
  darkMode: PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  darkMode: isDarkTheme(state),
});

export default connect(mapStateToProps)(Header);
