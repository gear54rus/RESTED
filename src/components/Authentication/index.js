import React, { PropTypes } from 'react';
import { connect } from 'react-redux';
import { FormSection, Field } from 'redux-form';
import { FormGroup, FormControl, Col, ControlLabel } from 'react-bootstrap';

import { textFieldShape } from 'propTypes/field';
import Collapsible from 'components/Collapsible';
import { authTypes } from 'store/auth/sagas';
import { getAuthType } from 'store/request/selectors';

import BasicFields from './BasicAuthFields';
import APSTokenFields from './APSTokenFields';
import OAuth1Fields from './OAuth1Fields';

function AuthTypeField({ input }) {
  return (
    <FormGroup controlId="auth">
      <Col
        componentClass={ControlLabel}
        xs={2}
      >
        Type
      </Col>
      <Col xs={8}>
        <FormControl
          componentClass="select"
          placeholder="Authentication type"
          {...input}
        >
          {Object.entries(authTypes).map(([type, { caption }]) => (
            <option
              key={type}
              value={type}
            >
              {caption}
            </option>
          ))}
        </FormControl>
      </Col>
    </FormGroup>
  );
}

AuthTypeField.propTypes = textFieldShape;

function Authentication({ authType }) {
  let authFields;

  switch (authType) {
    case 'basic':
      authFields = BasicFields;
      break;
    case 'apsToken':
      authFields = APSTokenFields;
      break;
    case 'oauth1':
      authFields = OAuth1Fields;
      break;
    default:
      authFields = null;
  }

  return (
    <Collapsible
      title="Authentication"
      id="auth"
    >
      <Field
        name="type"
        component={AuthTypeField}
      />
      {authFields && (
        <FormSection
          name={authType}
          component={authFields}
        />
      )}
    </Collapsible>
  );
}

Authentication.propTypes = {
  authType: PropTypes.oneOf(Object.keys(authTypes)),
};

const mapStateToProps = state => ({
  authType: getAuthType(state),
});

export default connect(mapStateToProps)(Authentication);
