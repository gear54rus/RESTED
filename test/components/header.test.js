import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import { Header } from 'components/Header';

test('Header renders correctly', () => {
  const tree = renderer.create(
    <Header />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Header contains the RESTED APS logo', () => {
  const tree = mount(
    <Header />,
  );
  expect(tree.find('img').prop('src')).toBe('img/rested-aps-logo.png');
});

test('Header contains the RESTED APS name', () => {
  const tree = mount(
    <Header />,
  );
  expect(tree.find('h1 span').prop('children')).toBe('RESTED APS');
});

