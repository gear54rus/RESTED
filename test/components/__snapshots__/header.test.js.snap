// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header
  className="cdHpBO"
>
  <h1>
    <a
      className="gRAapk"
      href="https://github.com/gear54rus/RESTED-APS"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        className="logo"
        height="40"
        role="presentation"
        src="img/rested-aps-logo.png"
      />
      <span>
        RESTED APS
      </span>
    </a>
  </h1>
  <a
    href="https://chrome.google.com/webstore/detail/rested-aps/omkndfeccmeplaimlpaefimnimmniccl"
    rel="noopener noreferrer"
    target="_blank"
  >
    Also available for Google Chrome
     
    <img
      alt="Chrome icon"
      src="https://upload.wikimedia.org/wikipedia/commons/a/a5/Google_Chrome_icon_%28September_2014%29.svg"
      style={
        Object {
          "height": "2em",
        }
      }
    />
  </a>
</header>
`;
